### DISCUSSION TITLE
How to write a wav file using a sequencer

### DISCUSSION THREAD

**Discussion by bramtayl** [FluidSynth/fluidsynth#1328] [Q&A]
Using the midi sequencer, I can generate and play simple tunes in real time. But I don't know how to write the tunes out as wav files.

I tried:

```
fluid_settings_setstr(fluid_settings_pointer, "audio.driver", "file");
fluid_settings_setstr(fluid_settings_pointer, "audio.file.name", "test.wav");
fluid_settings_setstr(fluid_settings_pointer, "player.timing-source", "sample");
fluid_settings_setint(fluid_settings_pointer, "synth.lock-memory", 0);
```

But I end up generating an empty wav file.

**Comment by derselbst** [comment]
How about https://www.fluidsynth.org/api/FileRenderer.html ?

**Comment by bramtayl** [comment]
I tried that but I don't have a midi file to read in. I'm generating the midi sequence with a program. From the documentation here:

https://www.fluidsynth.org/api/group__file__renderer.html

It says "If you are looking for a way to write audio generated from real-time events (for example from an external sequencer or a MIDI controller) to a file, please have a look at the file [Audio Driver](https://www.fluidsynth.org/api/group__audio__driver.html) instead."

**Comment by bramtayl** [comment]
Got it!
